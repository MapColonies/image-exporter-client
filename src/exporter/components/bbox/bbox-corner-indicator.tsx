import React, { useMemo } from 'react';
import { createStyles, makeStyles, Theme } from '@material-ui/core/styles';

interface BBoxCornerProps {
  corner: Corner;
  className?: string;
}

export enum Corner {
  TOP_RIGHT = 'TOP_RIGHT',
  BOTTOM_RIGHT = 'BOTTOM_RIGHT',
  BOTTOM_LEFT = 'BOTTOM_LEFT',
  TOP_LEFT = 'TOP_LEFT',
  UNKNOWN = 'UNKNOWN',
}

const useStyle = makeStyles((theme: Theme) =>
  createStyles({
    bbox: {
      overflow: 'hidden',
      position: 'relative',
      border: 'solid 1px',
      width: '60px',
      height: '56px',
    },

    bboxLeftBottomCorner: {
      "&::before": {
        margin: '-1em',
        borderRadius: '50%',
        position: 'absolute',
        padding: '1em',
        boxShadow: '0 0 7px #b53',
        background: '#95a',
        content: "''",
        bottom: 0,
        left: 0
      }
    },
    bboxLeftTopCorner: {
      "&::before": {
        margin: '-1em',
        borderRadius: '50%',
        position: 'absolute',
        padding: '1em',
        boxShadow: '0 0 7px #b53',
        background: '#95a',
        content: "''",
        top: 0,
        left: 0
      }
    },
    bboxRightTopCorner: {
      "&::before": {
        margin: '-1em',
        borderRadius: '50%',
        position: 'absolute',
        padding: '1em',
        boxShadow: '0 0 7px #b53',
        background: '#95a',
        content: "''",
        top: 0,
        right: 0
      }
    },
    bboxRightBottomCorner: {
      "&::before": {
        margin: '-1em',
        borderRadius: '50%',
        position: 'absolute',
        padding: '1em',
        boxShadow: '0 0 7px #b53',
        background: '#95a',
        content: "''",
        bottom: 0,
        right: 0
      }
    },

  })
);

const getCornerClass = (classes: Record<string,string>, cornerToIndicate: Corner): string => {
  switch(cornerToIndicate){
    case Corner.TOP_RIGHT:
      return classes.bboxRightTopCorner;
    case Corner.TOP_LEFT:
      return classes.bboxLeftTopCorner;
    case Corner.BOTTOM_RIGHT:
      return classes.bboxRightBottomCorner;
    case Corner.BOTTOM_LEFT:
      return classes.bboxLeftBottomCorner;
    default:
      return '';
  }
};

export const BBoxCorner: React.FC<BBoxCornerProps> = ({ corner, className }) => {
  const classes = useStyle();
  const bboxCorner = useMemo(() => getCornerClass(classes, corner), [classes, corner]);
  return (
    <div className={`${classes.bbox} ${bboxCorner} ${className ?? ''}`}></div>
  );
}
