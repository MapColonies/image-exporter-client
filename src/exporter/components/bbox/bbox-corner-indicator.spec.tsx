import React from 'react';
import { shallow } from 'enzyme';
import { BBoxCorner, Corner } from './bbox-corner-indicator';

const corner = Corner.BOTTOM_LEFT;

describe('BBoxCorner component', () => {
  it('renders correctly', () => {

    const wrapper = shallow(
      <BBoxCorner corner={corner} />
    );

    expect(wrapper).toMatchSnapshot();
  });

  it('renders as DIV', () => {
    const wrapper = shallow(
      <BBoxCorner corner={corner} />
    );

    const labelContainer = wrapper.find('div');
    expect(labelContainer).toBeTruthy(); 
  });

  it('marks appropriate corner', () => {
    const wrapper = shallow(
      <BBoxCorner corner={corner} />
    );

    const labelContainer = wrapper.find('div.bboxLeftBottomCorner');
    expect(labelContainer).toBeTruthy(); 
  });

  it('render DIV with marked corner and defined css className ', () => {
    const className = 'test';
    const wrapper = shallow(
      <BBoxCorner corner={corner} className={className}/>
    );

    const labelContainer = wrapper.find(`div.bboxLeftBottomCorner.${className}`);
    expect(labelContainer).toBeTruthy(); 
  });


});
