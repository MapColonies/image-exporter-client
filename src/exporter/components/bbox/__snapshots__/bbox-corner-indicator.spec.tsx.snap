// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BBoxCorner component renders correctly 1`] = `
<div
  className="makeStyles-bbox-1 makeStyles-bboxLeftBottomCorner-2 "
/>
`;
